### DISCUSSION TITLE
Dynamic layout

### DISCUSSION THREAD

**Discussion by lucas-a-meyer** [quarto-dev/quarto-cli#954] [Q&A] [answered]
Let's say I have 40 PNG figures representing 4 measures for 10 different countries. What I want to do is print the title of the figures, and then the four figures in a 2x2 matrix. Is there a way of doing this in a `for` loop in Jupyter?

The trouble I'm having is that if I define the cell layout as [[1], [1,1], [1,1]], the first country comes right, but instead of repeating the pattern of 1, [1,1], [1,1], Quarto continues the pattern with [1,1]s until the end, so the title of the second country takes the place that I expected to have the first figure.

```
#| layout: "[[1], [1,1], [1,1]]"
for name in country_names:
  display(Markdown(f"### Dependency Ratios for {name}"))
  for time in years:
    display(Markdown(f"![{name}-{time}](figures/pyr_{name}_{time}.png)"))
```

<img width="659" alt="image" src="https://user-images.githubusercontent.com/1373609/169855471-7e6aef98-9ab3-4ea7-a288-5001665cad18.png">

**Comment by jmbuhr** [comment, marked as answer]
If you want the headings to be programatically generated and all part of the same figure, you may have to resort to a nested layout and generating the subfigures programatically as well. You can inspect the markdown that quarto generates for a figure and recreate that from your python code (https://quarto.org/docs/computations/execution-options.html#intermediates)

**Reply by lucas-a-meyer** [reply]
Thanks!

I don't know how to mark this as a solution otherwise I would.

**Reply by eitsupi** [reply]
You need to switch the category to Q&A.
